File: Sorting/Quicksort/Quicksort.py
from Sorting.Quicksort.Partition import Partition

def Quicksort (A, p, r):
    if p < r:
        q = Partition (A, p, r)
        Quicksort (A, p, q - 1)
        Quicksort (A, q + 1, r)

File: Sorting/Quicksort/Java/Quicksort.java
package Sorting.Quicksort.Java;

public class Quicksort {
    public static void quicksort(int[] A, int p, int r) {
        if (p < r) {
            int q = Partition.partition(A, p, r);
            quicksort(A, p, q - 1);
            quicksort(A, q + 1, r);
        }
    }
}